qlib/finco/task.py

```diff
@@ -317,15 +312,30 @@ def execute(self):
         if isinstance(analysers, list) and len(analysers):
             self.logger.info(f"selected analysers: {analysers}", plain=True)
 
+            workflow_config = (
+                self._context_manager.get_context("workflow_config")
+                if self._context_manager.get_context("workflow_config")
+                else "workflow_config.yaml"
+            )
+            workspace = self._context_manager.get_context("workspace")
+            workflow_path = workspace.joinpath(workflow_config)
+            with workflow_path.open() as f:
+                workflow = yaml.safe_load(f)
+
+            experiment_name = workflow["experiment_name"] if "experiment_name" in workflow else "workflow"
+
+            model = init_instance_by_config(workflow["task"]["model"])
+            dataset = init_instance_by_config(workflow["task"]["dataset"])
+
             tasks = []
             for analyser in analysers:
                 if analyser in self.__ANALYZERS_PROJECT.keys():
                     tasks.append(
                         self.__ANALYZERS_PROJECT.get(analyser)(
-                            workspace=self._context_manager.get_context("workspace"),
-                            model=self._context_manager.get_context("model"),
-                            dataset=self._context_manager.get_context("dataset"),
-                            recorder=self._context_manager.get_context("recorder"),
+                            workspace=workspace,
+                            model=model,
```

Comment:
I don't think `SignalRecord` is one of Analysor.
Please refer to `SignalAnalyzer.__name__: SignalRecord,`
Thus `model` and `dataset` is not necessary paramters

Reply:
Analyser is independent of Recorder now